lib/Service/Provisioning/Manager.php

```diff
@@ -131,6 +141,19 @@ private function updateAccount(IUser $user, MailAccount $account, Config $config
 		$account->setOutboundHost($config->getSmtpHost());
 		$account->setOutboundPort($config->getSmtpPort());
 		$account->setOutboundSslMode($config->getSmtpSslMode());
+		$account->setSieveEnabled($config->getSieveEnabled());
+
+		if ($config->getSieveEnabled()) {
+			$account->setSieveUser($config->buildSieveUser($user));
+			$account->setSieveHost($config->getSieveHost());
+			$account->setSievePort($config->getSievePort());
+			$account->setSieveSslMode($config->getSieveSslMode());
+		} else {
+			$account->setSieveUser(null);
+			$account->setSieveHost(null);
+			$account->setSievePort(null);
+			$account->setSieveSslMode(null);
+		}
 
 		return $account;
 	}
```

Comment:
```suggestion
		if ($config->getSieveEnabled()) {
			$account->setSieveUser($config->buildSieveUser($user));
			$account->setSieveHost($config->getSieveHost());
			$account->setSievePort($config->getSievePort());
			$account->setSieveSslMode($config->getSieveSslMode());
			
			return $account;
		}
		
		$account->setSieveUser(null);
		$account->setSieveHost(null);
		$account->setSievePort(null);
		$account->setSieveSslMode(null);

		return $account;
	}
```

No need to wait to return here, if the condition is fulfilled, we can safely return the set values.

Reply:
yes and no :wink: 

I think this is one of those cases where the if and the else branch have equal importance. The early return would (to me) signal that it's some error condition or anything very specific.

I would keep the explicit if-else here.

But again, this formatting doesn't change any logic, nor does it make anything faster. I'm fine with either :)